File: client/client/DTO/Purchase/PurchaseResponseDto.cs
namespace client.DTO.Purchase;

public class PurchaseResponseDto
{
    public string Id { get; set; }
    public string CustomerId { get; set; }
    public string OfferId { get; set; }
    public int NumberOfItems { get; set; }
    public string status { get; set; }
    public DateTime purchaseTime { get; set; }
    public int PricePerItem { get; set; }
    public string BusinessName { get; set; }
}

File: server/src/main/java/com/example/server/dto/purchase/PurchaseResponseDto.java
package com.example.server.dto.purchase;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data @Builder @NoArgsConstructor @AllArgsConstructor public class PurchaseResponseDto
{

  @NotNull(message = "Purchase ID is required.") private String purchaseId;

  @NotNull(message = "Customer ID is required.") private String customerId;

  @NotNull(message = "Offer ID is required.") private String offerId;

  @NotNull(message = "Quantity is required.") private int numberOfItems;

  @NotNull(message = "Status is required.") private String status;

  @NotNull(message = "Purchase time is required.") private LocalDateTime purchaseTime;

  @NotNull(message = "Price per item is required.") private int pricePerItem;
  private String businessName;
  private String businessEmail;
  private String customerEmail;
}